vaccine_feed_ingest/runners/us/physicians_immediate/normalize.py

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env python
+
+import datetime
+import json
+import pathlib
+import re
+import sys
+from typing import Dict, List, Optional
+
+from vaccine_feed_ingest_schema import location as schema
+
+from vaccine_feed_ingest.utils.log import getLogger
+
+logger = getLogger(__file__)
+
+Site = Dict[str, str]
+
+
+def get_site_id(site: Site) -> str:
+    return f"us_physicians_immediate:{site['table_id']}_{site['row_id']}"
+
+
+def infer_address(site: Site) -> schema.Address:
```

Comment:
Take a look at https://pypi.org/project/usaddress/ for address parsing. I think that will help at lot here.

Reply:
This is really cool, I had no idea it existed. Thanks! Their README links to [usaddress-scourgify](https://github.com/GreenBuildingRegistry/usaddress-scourgify) which builds on usaddress but also claims to normalize the address according to post office standards. I just did a test run with scourgify, and got really good results but also a fair number of errors. Some of the egregious cases:

```
2021-05-12T04:28:41+0000 INFO:runners/us/physicians_immediate/normalize.py:Compare these two dicts:
Old method:
Address(street1='3111 S. Chicago South Milwaukee, WI 53172', street2=None, city=None, state='WI', zip='53172')
New method:
{'address_line_1': '3111 S',
 'address_line_2': None,
 'city': 'CHICAGO SOUTH MILWAUKEE',
 'postal_code': '53172',
 'state': 'WI'}
 2021-05-12T04:28:41+0000 INFO:runners/us/physicians_immediate/normalize.py:Compare these two dicts:
Old method:
Address(street1='1019 S. Green Bay Road Mount Pleasant, WI 53406', street2=None, city=None, state='WI', zip='53406')
New method:
{'address_line_1': '1019 S GREEN BAY ROAD MT',
 'address_line_2': None,
 'city': 'PLEASANT',
 'postal_code': '53406',
 'state': 'WI'}
2021-05-12T04:28:41+0000 INFO:runners/us/physicians_immediate/normalize.py:Compare these two dicts:
Old method:
Address(street1='2490 Bushwood Dr.Elgin, IL', street2=None, city=None, state='IL', zip=None)
New method:
{'address_line_1': '2490 BUSHWOOD',
 'address_line_2': None,
 'city': 'DRELGIN',
 'postal_code': None,
 'state': 'IL'}
```

When I can get a few minutes I'll dive into usaddress directly and see if I can do a bit better for this data.

Reply:
Hmmm. Those are fun errors. I can see why it made those mistakes, but not quite how to fix them.